tests/ui/allocator/empty-alloc-nonnull-guarantee.rs

```diff
@@ -1,3 +1,9 @@
+//! Test that empty allocations produce non-null pointers
+//!
+//! This test ensures that Rust's allocator maintains the invariant that
+//! Box<T> is always non-null, even for zero-sized types and empty allocations.
+//! This is crucial for memory safety guarantees in Rust.
```

Comment:
Nit: this guarantee is overly strong for what this test actually does, it's more like

> Check that the default global Rust allocator in combination with `Box` produces a non-null allocation even when the contained type is zero-sized.